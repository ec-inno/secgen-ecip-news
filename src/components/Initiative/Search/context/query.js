import React from 'react';

import { defaultLangKey } from '../../../../../languages';

const queryInit = {
  filters: {},
  status: 'ALL',
  language: defaultLangKey,
  pagination: '0/10',
};

const queryReducer = (state, action) => {
  switch (action.type) {
    case 'paginate': {
      return {
        ...state,
        pagination: action.pagination,
      };
    }

    case 'changeStatus': {
      return {
        ...state,
        pagination: queryInit.pagination,
        status: action.status,
      };
    }

    case 'changeLanguage': {
      return {
        ...state,
        pagination: queryInit.pagination,
        language: action.language,
      };
    }

    case 'changeFilter': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          [action.filter]: action.filterValue,
        },
      };
    }

    case 'setFilters': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          ...action.filters,
        },
      };
    }

    case 'removeFilter': {
      const { filters } = state;
      delete filters[action.filter];

      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...filters,
        },
      };
    }

    case 'reset':
    default: {
      return queryInit;
    }
  }
};

const Context = React.createContext();

export { queryInit, queryReducer, Context as default };
