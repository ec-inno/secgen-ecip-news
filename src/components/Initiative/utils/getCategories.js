const getCategories = t => {
  return [
    { value: 'any', label: t('All categories') },
    { value: 'AGRI', label: t('Agriculture') },
    { value: 'DEVCO', label: t('Aid and development cooperation') },
    { value: 'EURO', label: t('Business and economy') },
    { value: 'SANTE', label: t('Consumers and health') },
    { value: 'CULT', label: t('Culture and media') },
    { value: 'DECO', label: t('Digital economy and society') },
    { value: 'EDU', label: t('Education, youth and sport') },
    { value: 'EMPL', label: t('Employment and social affairs') },
    { value: 'ENER', label: t('Energy') },
    { value: 'ENV', label: t('Environment and climate') },
    { value: 'TRADE', label: t('External trade and relations') },
    { value: 'JUST', label: t('Justice and fundamental rights') },
    { value: 'MARE', label: t('Maritime affairs and fisheries') },
    { value: 'MIGR', label: t('Migration and asylum') },
    { value: 'REGIO', label: t('Regional development') },
    { value: 'RSH', label: t('Research') },
    { value: 'SEC', label: t('Security') },
    { value: 'TRA', label: t('Transport') },
    { value: 'REGFRA', label: t('REGFRA') },
    { value: 'REGECI', label: t('REGECI') },
    { value: 'COLLECI', label: t('COLLECI') },
    { value: 'EXAECI', label: t('EXAECI') },
  ];
};

export default getCategories;
