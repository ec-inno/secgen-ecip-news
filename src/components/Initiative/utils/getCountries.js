const getCountriesTranslated = t => {
  return [
    { value: 'bg', label: t('Bulgaria') },
    { value: 'es', label: t('Spain') },
    { value: 'cs', label: t('Czech Republic') },
    { value: 'da', label: t('Denmark') },
    { value: 'de', label: t('Germany') },
    { value: 'et', label: t('Estonia') },
    { value: 'el', label: t('Greece') },
    { value: 'fr', label: t('France') },
    { value: 'ga', label: t('Ireland') },
    { value: 'hr', label: t('Croatia') },
    { value: 'it', label: t('Italia') },
    { value: 'lv', label: t('Latvia') },
    { value: 'lt', label: t('Lithuania') },
    { value: 'hu', label: t('Hungary') },
    { value: 'mt', label: t('Malta') },
    { value: 'nl', label: t('Netherlands') },
    { value: 'pl', label: t('Poland') },
    { value: 'pt', label: t('Portugal') },
    { value: 'ro', label: t('Romania') },
    { value: 'sk', label: t('Slovakia') },
    { value: 'sl', label: t('Slovenia') },
    { value: 'fi', label: t('Finland') },
    { value: 'sv', label: t('Sweden') },
  ];
};

export default getCountriesTranslated;
