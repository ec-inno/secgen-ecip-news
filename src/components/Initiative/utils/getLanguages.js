const getLanguagesTranslated = t => {
  return [
    { value: 'bg', label: t('Bulgarian') },
    { value: 'es', label: t('Spanish') },
    { value: 'cs', label: t('Czech') },
    { value: 'da', label: t('Dansk') },
    { value: 'de', label: t('German') },
    { value: 'et', label: t('Estonian') },
    { value: 'el', label: t('Greek') },
    { value: 'en', label: t('English') },
    { value: 'fr', label: t('French') },
    { value: 'ga', label: t('Irish') },
    { value: 'hr', label: t('Croatian') },
    { value: 'it', label: t('Italian') },
    { value: 'lv', label: t('Latvian') },
    { value: 'lt', label: t('Lithuanian') },
    { value: 'hu', label: t('Hungarian') },
    { value: 'mt', label: t('Maltese') },
    { value: 'nl', label: t('Dutch') },
    { value: 'pl', label: t('Polish') },
    { value: 'pt', label: t('Portuguese') },
    { value: 'ro', label: t('Romanian') },
    { value: 'sk', label: t('Slovak') },
    { value: 'sl', label: t('Slovene') },
    { value: 'fi', label: t('Finnish') },
    { value: 'sv', label: t('Swedish') },
  ];
};

export default getLanguagesTranslated;
