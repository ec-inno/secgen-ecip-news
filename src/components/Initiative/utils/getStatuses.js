// These are values for filters, not resource path!
// Resource path values are LATEST, ONGOING, ANSWERED, ALL, REFUSED
const getStatuses = t => {
  return [
    { value: 'ALL', label: t('All') },
    { value: 'OPEN', label: t('Open') },
    { value: 'SUCCESSFUL', label: t('Successful') },
    { value: 'ARCHIVED', label: t('Archived') },
  ];
};

export default getStatuses;
