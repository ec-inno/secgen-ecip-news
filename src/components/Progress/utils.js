export const steps = [
  'REGISTERED',
  'COLLECTION_START_DATE',
  'ONGOING',
  'COLLECTION_EARLY_CLOSURE',
  'CLOSED',
  'VERIFICATION',
  'SUBMITTED',
  'ANSWERED',
];

/**
 * Reorder progress stages to match order of `steps`.
 * @param {Array<Object>} progress
 * @param {String} progress.name
 * @param {Boolean} progress.active
 * @param {Date} progress.date
 */
export const getStages = progress => {
  const stages = [];

  steps.forEach(step => {
    const match = progress.find(item => item.name === step);
    if (match) stages.push(match);
  });

  return stages;
};
