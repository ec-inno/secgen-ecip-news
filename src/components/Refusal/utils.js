import messages from './messages.json';

export const reasonsSupported = [
  'reason.action.registration.reject.abuse',
  'reason.action.registration.reject.competences',
  'reason.action.registration.reject.requirements',
  'reason.action.registration.reject.values',
];

export const getTranslation = ({ code, language }) =>
  messages.find(
    message => message.LNG_CODE === language && message.RTR_KEY === code
  );

export const getReasons = ({ input, language }) => {
  const output = [];

  input.forEach(reason => {
    if (reasonsSupported.includes(reason)) {
      const translation = getTranslation({ code: reason, language });
      if (translation) {
        output.push(translation);
      }
    }
  });

  return output;
};
