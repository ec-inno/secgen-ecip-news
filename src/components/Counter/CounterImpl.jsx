import React from 'react';

import { useI18nContext } from '@eci/context/I18n';
import { useCountersApi } from '@eci/utils/useCountersApi';

import Counter from './Counter';

const CounterImpl = () => {
  const { locale } = useI18nContext();
  const { response } = useCountersApi(locale);

  if (!response || !response.data) return '';

  return <Counter {...response.data} />;
};

export default CounterImpl;
