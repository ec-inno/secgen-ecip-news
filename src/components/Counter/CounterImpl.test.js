import React from 'react';
import renderer from 'react-test-renderer';

import * as I18nContext from '@eci/context/I18n';
import * as useCountersApi from '@eci/utils/useCountersApi';
import CounterImpl from './CounterImpl';

describe('CounterImpl', () => {
  it('does not show on lack of data', () => {
    jest.spyOn(I18nContext, 'useI18nContext').mockImplementation(() => ({
      locale: 'en',
    }));

    jest.spyOn(useCountersApi, 'useCountersApi').mockImplementation(() => ({
      response: {},
    }));

    const tree = renderer.create(<CounterImpl />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('can render statistics', () => {
    jest.spyOn(I18nContext, 'useI18nContext').mockImplementation(() => ({
      locale: 'en',
    }));

    jest.spyOn(useCountersApi, 'useCountersApi').mockImplementation(() => ({
      response: {
        data: {
          requests: '1',
          registered: '2',
          successful: '3',
        },
      },
    }));

    const tree = renderer.create(<CounterImpl />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
