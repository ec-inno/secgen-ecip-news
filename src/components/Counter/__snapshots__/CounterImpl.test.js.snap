// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterImpl can render statistics 1`] = `
<p
  className="ecl-u-type-paragraph ecl-u-mt-l ecl-u-d-flex ecl-u-justify-content-between"
>
  <span
    className="ecl-u-d-flex ecl-u-align-items-center"
  >
    <svg
      aria-hidden="true"
      className="ecl-u-mr-xs ecl-icon ecl-icon--m"
      focusable="false"
    >
      
      
      <use
        xlinkHref="file#general--file"
      />
    </svg>
    1
     
    registration requests
  </span>
  <span
    className="ecl-u-d-flex ecl-u-align-items-center"
  >
    <svg
      aria-hidden="true"
      className="ecl-u-mr-xs ecl-icon ecl-icon--m"
      focusable="false"
    >
      
      
      <use
        xlinkHref="file#general--edit"
      />
    </svg>
    2
     
    initiatives registered
  </span>
  <span
    className="ecl-u-d-flex ecl-u-align-items-center"
  >
    <svg
      aria-hidden="true"
      className="ecl-u-mr-xs ecl-icon ecl-icon--m"
      focusable="false"
    >
      
      
      <use
        xlinkHref="file#ui--check-filled"
      />
    </svg>
    3
     
    successful initiatives
  </span>
</p>
`;

exports[`CounterImpl does not show on lack of data 1`] = `""`;
